### PR TITLE
core: channel tracing to log lb policy changes

### PR DESCRIPTION
**Pull request by dapengzhang0**

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java [hunk 1]
```diff
@@ -43,11 +47,19 @@
   static final String GRPCLB_LOAD_BALANCER_FACTORY_NAME =
       "io.grpc.grpclb.GrpclbLoadBalancerFactory";
 
-  AutoConfiguredLoadBalancerFactory() {}
+  @Nullable
+  private final ChannelTracer channelTracer;
+  @Nullable
+  private final TimeProvider timeProvider;
+
+  AutoConfiguredLoadBalancerFactory(ChannelTracer channelTracer, TimeProvider timeProvider) {
```
Comment (zpencer) [inline core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java]: nit: use `@Nullable` here as well to be consistent

File: core/src/main/java/io/grpc/PickFirstBalancerFactory.java [hunk 2]
```diff
@@ -120,6 +120,11 @@ public void shutdown() {
         subchannel.shutdown();
       }
     }
+
+    @Override
+    public String toString() {
+      return "PickFirstBalancer";
```
Comment (zpencer) [inline core/src/main/java/io/grpc/PickFirstBalancerFactory.java]: If we don't define this explicitly, then we can differentiate between different instances. Would that be something we want?

If we really only want to show only the class name, can we get it via something like `getClass().getSimpleName()`?

Comment (dapengzhang0) [reply core/src/main/java/io/grpc/PickFirstBalancerFactory.java]: Done.

File: core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java (hunk 1 above)
Comment (dapengzhang0) [reply core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java]: Done.

**Comment by dapengzhang0** [comment]
PTALA

**Comment by dapengzhang0** [comment]
@zpencer Friendly ping

**Pull request merged**